tests/core/stdlib/buildid_test.go

```diff
@@ -1,9 +1,80 @@
-package main
+// +build go1.10
+
+/* Copyright 2018 The Bazel Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package buildid_test
 
 import (
+	"bytes"
+	"errors"
+	"os"
+	"os/exec"
+	"path/filepath"
 	"testing"
 )
 
-func TestEmptyBuildId(t *testing.T) {
-	// TODO: run go tool buildid on .a stdlib files to check it is empty
+func TestEmptyBuildID(t *testing.T) {
+	// Locate fmt.a (any .a file in the stdlib will do) and the buildid tool.
+	// The path may vary depending on the platform and architecture, so we
+	// just do a search.
+	var fmtPath, buildidPath string
+	done := errors.New("done")
+	var visit filepath.WalkFunc
+	visit = func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if (info.Mode() & os.ModeType) == os.ModeSymlink {
+			path, err = filepath.EvalSymlinks(path)
+			if err != nil {
+				return err
+			}
+			return filepath.Walk(path, visit)
+		}
+		if filepath.Base(path) == "fmt.a" {
```

Comment:
unfortunately `fmt` is not a good candidate because pure go packages where producing the same build ids.

that said, we should check 3 types of packages:
- the ones pure go (`fmt` looks like one)
- the ones with asm (`crypto/*` are in that case I think)
- the ones with cgo (`os/user` comes to mind)

You can find the list at https://github.com/bazelbuild/rules_go/issues/1391#issuecomment-377991295, perhaps check `fmt` and the ones from the list ?

Reply:
it's likely it'll be the same but for the sake of correctness and future proofing...

Reply:
also, we should probably check for the sandbox path by binary search for `bazel-sandbox` for cgo packages because of https://github.com/bazelbuild/rules_go/pull/1440/commits/f4201a501c6ca245054035fc54f8037715f241c0

but that may out of the scope of this PR though

Reply:
Very good advice. I've expanded the test to check `fmt.a`, `crypto/aes.a`, and `runtime/cgo.a`. That should cover those cases.

About searching for `bazel-sandbox`, `tests/legacy/reproducible_binary` checks for common reproducibility problems in compiled binaries, but I wrote it before `bazel-sandbox` appeared it paths, so it should be updated (in another PR).